pulsar-client-cpp/lib/BatchMessageContainer.cc

```diff
@@ -102,10 +102,18 @@ void BatchMessageContainer::sendMessage(FlushCallback flushCallback) {
     producer_.encryptMessage(impl_->metadata, impl_->payload, encryptedPayload);
     impl_->payload = encryptedPayload;
 
-    if (impl_->payload.readableBytes() > Commands::MaxMessageSize) {
-        // At this point the compressed batch is above the overall MaxMessageSize. There
-        // can only 1 single message in the batch at this point.
-        batchMessageCallBack(ResultMessageTooBig, messagesContainerListPtr_, nullptr);
+    ClientConnectionPtr cnx = producer_.getCnx().lock();
+    if (cnx) {
+        if (impl_->payload.readableBytes() > cnx->getMaxMessageSize()) {
+            // At this point the compressed batch is above the overall MaxMessageSize. There
+            // can only 1 single message in the batch at this point.
+            batchMessageCallBack(ResultMessageTooBig, messagesContainerListPtr_, nullptr);
+            clear();
+            return;
+        }
+    } else {
+        LOG_DEBUG("Connection not ready for batch message container.")
+        batchMessageCallBack(ResultNotConnected, messagesContainerListPtr_, nullptr);
```

Comment:
This is not correct. We need to be able to enqueue messages even when we're disconnected from the broker. 

One approach here could be to update the max-message size once we connect and then keep using the same value (until a new reconnection succeeds).